jekyll/_cci2/deployment-examples.md

```diff
@@ -0,0 +1,678 @@
+```
+version: 2
+
+workflows:
+  version: 2
+  build-deploy:
+    jobs:
+      - build-job
+      - deploy-job:
+          requires:
+            - build-job # Only run deploy job once build job has completed
+          filters:
+            branches:
+              only: master # Only run deploy job when commit is on the master branch
+
+jobs:
+  #  build and test jobs go here - not included for brevity
+  deploy-job:
+    docker:
+      - image: <docker-image-name-tag>
+    working_directory: ~/repo
+    steps:
+      - checkout
+      - run:
+          name: Bundle Install
+          command: bundle check || bundle install
+      - run:
+          name: Deploy if tests pass and branch is Master
+          command: bundle exec cap production deploy
+```
+
+## Cloud Foundry
+
+CircleCI has developed a CloudFoundry Orb that you can use to simplify your configuration workflows. The Cloud Foundry page in the [Orbs Registry](https://circleci.com/orbs/registry/orb/circleci/cloudfoundry) contains several different examples of how you can perform tasks with CloudFoundry, including the example below that shows how you can build and run blue green deployment in a single job - in this example `domain` will automatically be prefixed with `dark` and `live` for two subdomains to be specified. Validation steps would also need to be provided to allow the live deployment to go ahead.
+
+```
+version: 2.1
+
+orbs:
+  cloudfoundry: circleci/cloudfoundry@x.y.z # Use the Cloud Foundry orb in your config
+  
+workflows:
+  build_deploy:
+    jobs:
+      - cloudfoundry/blue_green:
+          appname: <your-app-name>
+          build_steps:
+            - run: echo 'your build steps'
+            - run: echo 'you can have more, too'
+            - run: echo 'or provide a workspace'
+          context: your-context
+          domain: your-domain
+          manifest: null
+          org: your-org
+          package: null
+          space: your-space
+          validate_steps: 
+            # Call any orbs or custom commands that validate the health of deployed application before letting Green deploy/reroute proceed.
+            # For example,  hitting a /health endpoint with curl and making sure the dark URL returns a 200.
+```
+
+If you would like more detailed information about various CloudFoundry orb elements that you can use in your configuration workflows, refer to the [CloudFoundry Orb](https://circleci.com/orbs/registry/orb/circleci/cloudfoundry) page in the [CircleCI Orbs Registry](https://circleci.com/orbs/registry/).
+
+### Deploy to Cloud Foundry with 2.0 Config
+
+Cloud Foundry deployments require the Cloud Foundry CLI. Be sure to match the architecture to your Docker image (the commands below assume you are using a Debian-based image). This example pattern implements "Blue-Green" deployments using Cloud Foundry's map-route/unmap-route commands, which is an optional feature above and beyond a basic `cf push`.
+
+#### Install the CLI
```

Comment:
I would much rather display this as one full config rather than snippet sections - but I don't feel confident to put it together myself. Any help appreciated!

Reply:
Yeah just show a full config example and talk about it - I think it's easier to follow that way but that's just me, I am happy to be wrong :-)